instantout/store.go

```diff
@@ -0,0 +1,432 @@
+		},
+	)
+}
+
+// GetInstantLoopOut returns the instant loop out for the given swap
+// hash.
+func (s *SQLStore) GetInstantLoopOut(ctx context.Context, swapHash []byte) (
+	*InstantOut, error) {
+
+	row, err := s.baseDb.GetInstantOutSwap(ctx, swapHash)
+	if err != nil {
+		return nil, err
+	}
+
+	updates, err := s.baseDb.GetInstantOutSwapUpdates(ctx, swapHash)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.sqlInstantOutToInstantOut(ctx, row, updates)
+}
+
+// ListInstantLoopOuts returns all instant loop outs that are in the
+// store.
+func (s *SQLStore) ListInstantLoopOuts(ctx context.Context) ([]*InstantOut,
+	error) {
+
+	rows, err := s.baseDb.GetInstantOutSwaps(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var instantOuts []*InstantOut
+	for _, row := range rows {
+		updates, err := s.baseDb.GetInstantOutSwapUpdates(
+			ctx, row.SwapHash,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		instantOut, err := s.sqlInstantOutToInstantOut(
+			ctx, sqlc.GetInstantOutSwapRow(row), updates,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		instantOuts = append(instantOuts, instantOut)
+	}
+
+	return instantOuts, nil
+}
+
+// sqlInstantOutToInstantOut converts sql rows to an instant out struct.
+func (s *SQLStore) sqlInstantOutToInstantOut(ctx context.Context,
+	row sqlc.GetInstantOutSwapRow, updates []sqlc.InstantoutUpdate) (
+	*InstantOut, error) {
+
+	swapHash, err := lntypes.MakeHash(row.SwapHash)
+	if err != nil {
+		return nil, err
+	}
+
+	swapPreImage, err := lntypes.MakePreimage(row.Preimage)
+	if err != nil {
+		return nil, err
+	}
+
+	serverKey, err := btcec.ParsePubKey(row.SenderScriptPubkey)
+	if err != nil {
+		return nil, err
+	}
+
+	clientKey, err := btcec.ParsePubKey(row.ReceiverScriptPubkey)
+	if err != nil {
+		return nil, err
+	}
+
+	var finalizedHtlcTx *wire.MsgTx
+	if row.FinalizedHtlcTx != nil {
+		finalizedHtlcTx = &wire.MsgTx{}
+		err := finalizedHtlcTx.Deserialize(bytes.NewReader(
+			row.FinalizedHtlcTx,
+		))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var finalizedSweepLessSweepTx *wire.MsgTx
+	if row.FinalizedSweeplessSweepTx != nil {
+		finalizedSweepLessSweepTx = &wire.MsgTx{}
+		err := finalizedSweepLessSweepTx.Deserialize(bytes.NewReader(
+			row.FinalizedSweeplessSweepTx,
+		))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var sweepTxHash *chainhash.Hash
+	if row.SweepTxHash != nil {
+		sweepTxHash, err = chainhash.NewHash(row.SweepTxHash)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var outgoingChanSet loopdb.ChannelSet
+	if row.OutgoingChanSet != "" {
+		outgoingChanSet, err = loopdb.ConvertOutgoingChanSet(
+			row.OutgoingChanSet,
+		)
+		if err != nil {
+			return nil, err
+		}
+	}
+	reservationIds, err := byteSliceToReservationIds(row.ReservationIds)
+	if err != nil {
+		return nil, err
+	}
+
+	reservations := make([]*reservation.Reservation, 0, len(reservationIds))
+	for _, id := range reservationIds {
+		reservation, err := s.reservationStore.GetReservation(
```

Comment:
nit: i'd move this out of the `sqlInstantOutToInstantOut` function to where we call it instead so that this function doesn't do any DB calls.